Small fixes and a big documentation update (#29896)
dimagi/commcare-hq: corehq/apps/couch_sql_migration/README.md

The commands:

```sh
./manage.py couch_domain_report --help
./manage.py migrate_multiple_domains_from_couch_to_sql --help
./manage.py migrate_domain_from_couch_to_sql --help
./manage.py couch_sql_diff --help
```

## Setup

First, from a machine where you have `commcare-cloud` installed, check if there
are any Couch domains that need to be migrated:

```sh
commcare-cloud <env> django-manage couch_domain_report
```

You can stop here if the total at the bottom of the report is zero. Your
environment does not need to be migrated. On the other hand, if the total is
non-zero, proceed to setup and run migrations.

The migration commands outlined in this document are typically run on a "limited
release" on a machine in your CommCareHQ cluster, usually the `django_manage`

> machine. 8GB of RAM is recommended, although less may be necessary for small
> domains. It is also recommended to use a screen session so that network

Reviewer: nit: minor reword + commcare-cloud has builtin support for tmux but not for screen

```suggestion
machine. 8GB of RAM is recommended, although small domains may require less.
It is also recommended to use a tmux or screen session so that network
```

disruptions between your workstation and the `django_manage` machine do not

> interrupt migration processes. Proceed to setup a limited release and screen

Reviewer: ```suggestion
interrupt migration processes. Proceed to setup a limited release and tmux or screen
```

session:

```sh
ENV=<your-env-name>

commcare-cloud $ENV fab setup_limited_release:keep_days=30
# make a note of the release directory path

# start a screen session
commcare-cloud $ENV ssh django_manage -t screen -S forms-migration

# if the connection drops, use the following command to reconnect
commcare-cloud $ENV ssh django_manage -t screen -x forms-migration

# in the screen session
sudo -iu cchq
export RELEASE_PATH=...  # set to the limited release path created above
export CCHQ_MIGRATION_STATE_DIR=~/forms-migration  # customize path as desired
mkdir $CCHQ_MIGRATION_STATE_DIR

> cd $RELEASE_PATH

Reviewer: most of this can be moved if you use commcare-cloud's tmux support. Here's a oneline command to run the whole thing in a tmux session:

```
cchq <env> django-manage --tmux couch_domain_report --output-dir=/home/cchq/forms-migration

Author: `CCHQ_MIGRATION_STATE_DIR` is required by the other migration commands, so it must be set or an extra command line argument is required.

I do not believe the command you provided will create the output directory, so I don't think that will work.
